src/commcare_cloud/ansible/deploy_formplayer.yml

```diff
@@ -25,3 +25,19 @@
       tags: kinesis-agent
   handlers:
     - include: roles/monit/handlers/main.yml
+
+- name: Cleanup formplayer releases
+  hosts: formplayer
+  tasks:
+    - name: List of releases
+      find:
+        paths="{{ formplayer_build_dir }}"
+        file_type=directory
+      register: releases
+
+    - name: Keep the last 2 releases
+      file:
+        path="{{ item.path }}"
+        state=absent
+      with_items:
+        - "{{ (releases.files | sort(attribute='ctime'))[:-2] }}"
```

Comment:
I'd suggest that we do two more things here:
- move the cleanup _before_ we create a new release
- resolve the 'current' directory's symlink and make sure that that directory is excluded from the list of directories to be removed.

This will result in us always knowing that the previous release is there to roll back to even if it's not chronologically the most recent release. That could happen if a directory is created for a release but there's an error and it exits before updating the symlink. The deploy process is designed that way, so that if there's an error during the deploy, it will effectively be a no-op not affecting the running environment at all. In such a case though you might have

- release 1
- release 2 <-- current
- release 3 (failed)
- release 4 (failed)
- release 5 (failed)

and you wouldn't want the next deploy to only save release 5. With my suggestion it would save release 5, release 4 (previous 2) and release 3 (current release), and then create the next release as well.

Reply:
how would we know if a release directory was from a failed deploy or not? In HQ we keep a metadata file of all the successful releases. We could do something similar with formplayer.